docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,271 @@
+# Integration Testing
+
+## Introduction
+
+This article motivates developers to adopt integration testing by explaining how to write, run,
+and evaluate the results of integration tests.
+
+## Quick reference
+
+1. [colcon](https://github.com/ros2/ros2/wiki/Colcon-Tutorial) is used to build and run tests.
+2. [launch testing](https://github.com/ros2/launch/tree/master/launch_testing) launches nodes and runs tests.
+3. [Testing in general](testing-in-general.md) describes the big picture of testing.
+
+## Integration testing
+
+An integration test is defined as the phase in software testing where individual software
+modules are combined and tested as a group. Integration tests occur after unit tests, and before
+validation tests.
+
+The input to an integration test is a set of independent modules that have been unit tested. The set
+of modules is tested against the defined integration test plan, and the output is a set of
+properly integrated software modules that is ready for system testing.
+
+## Value of integration testing
+
+Integration tests determine if independently developed software modules work correctly
+when the modules are connected to each other. In ROS 2, the software modules are called
+nodes. As a special case, testing a single node can be referred to as component testing.
+
+Integration tests help to find the following types of errors:
+
+- Incompatible interaction between nodes, such as non-matching topics, different message types, or incompatible QoS settings
+- Reveal edge cases that were not touched with unit tests, such as a critical timing issue, network communication delay, disk I/O failure, and many other problems that can occur in production environments
+- Using tools like `stress` and `udpreplay`, performance of nodes is tested with real data or while the system is under high CPU/memory load, where situations such as `malloc` failures can be detected
```

Comment:
Hmm. This one is a bit more tricky, so please let me know if reordering the sentence maintains the intended meaning.

```suggestion
- Issues that can occur while the system is under high CPU/memory load, such as `malloc` failures. This can be test using tools like `stress` and `udpreplay` to test the performance of nodes with real data.```